libsolidity/interface/CompilerStack.cpp

```diff
@@ -1126,11 +1126,15 @@ void CompilerStack::generateIR(ContractDefinition const& _contract)
 	if (!compiledContract.yulIR.empty())
 		return;
 
+	string dependenciesSource;
 	for (auto const* dependency: _contract.annotation().contractDependencies)
```

Comment:
`contractDependencies` is not the right thing here. I think we should only add the `yulIR` to the object once we hit the `::Creation` call. And we should also add it as a named sub-object, not just concatenate the code.

Reply:
Sorry, this was not clear enough: I think we need to generate all the code for all contracts in `_contract.annotation().contractDependencies` before we start generating the code for the current contract, but we should not just blindly concatenate the code.

Reply:
Instead, IRGenerator should somehow get access to `m_contracts.at(dependency->fullyQualifiedName()).yulIR`

Reply:
Can you add a commandline test that shows which code is added where?

Reply:
I added commandline test for this, it will help to finish this dependency part